### PR TITLE
http request handler

### PR DESCRIPTION
**Pull request by dawidrogowicz**
http-request-handler and added withId to jsonApi.post

### CODE REVIEW

File: packages/http-request-handler/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+Http request handler can be used with HttpClient to handle request, using Node.js native http/https modules.
+
+## Quick Start
+
+Add the Http Request Handler package, assuming that you already have `@coolio/http` installed:
+
+```bash
+npm install @coolio/http-request-handler
+```
```
Comment (krzysztof-miemiec) [inline packages/http-request-handler/README.md]: Can you add TS doc also?
```typescript
import { httpRequestHandler } from '@coolio/http-request-handler';

const client = new HttpClient(
 //...
  requestHandler: httpRequestHandler({
    // Any options here?
  }),
);
```

File: packages/http-request-handler/package.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "@coolio/http-request-handler",
+  "version": "0.8.3",
+  "description": "Http request handler for Coolio",
+  "keywords": [
+    "auth",
```
Comment (krzysztof-miemiec) [inline packages/http-request-handler/package.json]: `"http", "request-handler", "request", "handler", "coolio"`

File: packages/http-request-handler/src/createSimpleServer.helper.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import http from 'http';
```
Comment (krzysztof-miemiec) [inline packages/http-request-handler/src/createSimpleServer.helper.ts]: I believe it belongs to spec/testing part of `@coolio/http` - import it from there (can be a direct `@coolio/http/dist/createSimpleServer.helper` import)

File: packages/http-request-handler/src/createSimpleServer.helper.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+export interface SimpleServer {
+  port: number;
+  host: string;
+  fullAddress: string;
+  close: () => Promise<void>;
+}
+
+export const createSimpleServer = ({
```
Comment (krzysztof-miemiec) [inline packages/http-request-handler/src/createSimpleServer.helper.ts]: Can you add a `// TODO` for me to just use `express` for testing in the future? 😅

File: packages/http-request-handler/src/helpers/headers.helper.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { HttpHeaders } from '@coolio/http';
+
+export const getHeader = (headers: HttpHeaders | undefined, header: string): string | undefined => {
```
Comment (krzysztof-miemiec) [inline packages/http-request-handler/src/helpers/headers.helper.ts]: It belongs to `@coolio/http` - import it from there.

File: packages/http-request-handler/src/httpResponseHeaders.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { getHeader, parseHeaders } from './helpers';
+import { HttpHeaders } from '@coolio/http';
+import { IncomingHttpHeaders } from 'http';
+
+export class HttpResponseHeaders {
```
Comment (krzysztof-miemiec) [inline packages/http-request-handler/src/httpResponseHeaders.ts]: It belongs to `@coolio/http` package - you should use it from there.

File: packages/http-request-handler/src/helpers/spec/headers.helper.spec.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import { getHeader } from '../headers.helper';
```
Comment (krzysztof-miemiec) [inline packages/http-request-handler/src/helpers/spec/headers.helper.spec.ts]: It belongs to `@coolio/http` - import it from there.

File: packages/http-request-handler/src/helpers/encoder.helper.ts [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import TypedArray = NodeJS.TypedArray;
```
Comment (krzysztof-miemiec) [inline packages/http-request-handler/src/helpers/encoder.helper.ts]: It belongs to `@coolio/http` - import it from there.

File: packages/http/src/testing/createSimpleServer.helper.ts [hunk 9]
```diff
@@ -61,4 +62,3 @@ export const createSimpleServer = ({
     close: () => new Promise((resolve, reject) => server.close((err: any) => err ? reject(err) : resolve())),
   };
 };
-
```
Comment (krzysztof-miemiec) [inline packages/http/src/testing/createSimpleServer.helper.ts]: Where's newline?

File: tsconfig.esm.json [hunk 10]
```diff
@@ -1,8 +1,9 @@
 {
   "extends": "./tsconfig.json",
   "compilerOptions": {
-    "module": "esnext",
+    "module": "commonjs",
```
Comment (krzysztof-miemiec) [inline tsconfig.esm.json]: Change it to `esnext`. You can `import` `@coolio/http` from `dist/cjs` if esnext doesn't work correctly. In the future I'll try to export these in more user-friendly manner.

File: packages/http/src/testing/createSimpleServer.helper.ts (hunk 9 above)
Comment (dawidrogowicz) [reply packages/http/src/testing/createSimpleServer.helper.ts]: obraz
🤔

Comment (krzysztof-miemiec) [reply packages/http/src/testing/createSimpleServer.helper.ts]: Ok, so that was a duplicate trailing newline. 👍

**Pull request merged**